### PR TITLE
review:Add a class NopolStatus for NopolResult

### LINKED ISSUE #126: Create a NopolStatus to give information about NopolResult and print it in Main before leaving
- Timeout on the whole block
- No angelic value
- No synthesis 
- Patch

### PR DESCRIPTION
**Pull request by surli**
Fix #126

### CODE REVIEW

**Comment by surli** [comment]
I started the PR to close #126 but the problem is: the status we get will only represent the state of the patch creation for the last statement, and not for the whole process. 
Then should I keep a map representing the status for each statement and infer from it a global status for NopolResult?

File: nopol/src/main/java/fr/inria/lille/repair/nopol/NoPol.java [hunk 1]
```diff
@@ -226,6 +248,8 @@ public Object call() throws Exception {
 
 		if (angelicValue != null) {
 			this.nopolResult.incrementNbAngelicValues();
+		} else {
+			this.nopolResult.setNopolStatus(NopolStatus.NO_ANGELIC_VALUE);
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/repair/nopol/NoPol.java]: Unnecessary statement

File: nopol/src/main/java/fr/inria/lille/repair/nopol/NoPol.java [hunk 2]
```diff
@@ -237,17 +261,22 @@ public Object call() throws Exception {
...
-				if (nopolContext.isOnlyOneSynthesisResult()) {
-					return patches;
+
+		if (tmpPatches.size() > 0) {
+			for (int i = 0; i < tmpPatches.size(); i++) {
+				Patch patch = tmpPatches.get(i);
+				if (nopolContext.isSkipRegressionStep() || isOk(patch, tests, synth.getProcessor())) {
+					patches.add(patch);
+					if (nopolContext.isOnlyOneSynthesisResult()) {
+						this.nopolResult.setNopolStatus(NopolStatus.PATCH);
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/repair/nopol/NoPol.java]: Unnecessary statement

File: nopol/src/main/java/fr/inria/lille/repair/Main.java [hunk 3]
```diff
@@ -111,6 +114,12 @@ public Object call() throws Exception {
 			e.printStackTrace();
 			main.showUsage();
 		}
+		if (result != null) {
+			System.out.println(result.getNopolStatus());
+
+			returnCode = (result.getPatches().isEmpty()) ? 1 : 0;
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/repair/Main.java]: status code 1 means: "Miscellaneous errors, such as "divide by zero" and other impermissible operations"
But it is not the case

**Pull request merged**